test/e2e/e2e_test.go

```diff
@@ -690,6 +683,88 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQueryForVector(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (model.Vector, error) {
+	results, warnings, err := promClient.Query(context.Background(), query, sampleTime)
+	if err != nil {
+		t.Errorf("failed to execute prometheus query: %v", err)
```

Comment:
Why is desirable to both log and return the error?

Reply:
Yes, you are right! maybe because of my habit of over-logging things. Modified it to return the error now instead of printing it.